pkg/lwip/contrib/lwip.c

```diff
@@ -98,12 +98,15 @@
 #define LWIP_NETIF_NUMOF        ARRAY_SIZE(socket_zep_params)
 #endif
 
-#ifdef MODULE_ESP_ETH      /* is mutual exclusive with above ifdef */
-#define LWIP_NETIF_NUMOF        (1)
-#endif
-
-#ifdef MODULE_ESP_WIFI     /* is mutual exclusive with above ifdef */
-#define LWIP_NETIF_NUMOF        (1)
+/* is mutual exclusive with above ifdef */
+#if (defined(MODULE_ESP_ETH) || defined(MODULE_ESP_WIFI))
+# if (defined(MODULE_ESP_ETH) && defined(MODULE_ESP_WIFI))
+#  define LWIP_NETIF_NUMOF        (2)
+#  define ESP_WIFI_INDEX          (1)
+# else
+#  define LWIP_NETIF_NUMOF        (1)
+#  define ESP_WIFI_INDEX          (0)
+# endif
 #endif
```

Comment:
```suggestion
#if IS_USED(MODULE_ESP_ETH) && IS_USED(MODULE_ESP_WIFI)
#define LWIP_NETIF_NUMOF        (2)
#define ESP_WIFI_INDEX          (1)
#elif IS_USED(MODULE_ESP_ETH) || IS_USED(MODULE_ESP_WIFI)
#define LWIP_NETIF_NUMOF        (1)
#define ESP_WIFI_INDEX          (0)
#endif
```

This way you do not need to have that many nested CPP instructions. Also note the usage of `IS_USED()` which is preferred to declare module usage.